int32 device
int32 cmd1
int32 cmd0
